keras_cv/layers/object_detection/roi_generator.py

```diff
@@ -0,0 +1,181 @@
+    1) pre_nms_topk scores and boxes sorted and selected per level
+    2) nms applied and selected post_nms_topk scores and rois per level
+    3) combined scores and rois across all levels
+    4) post_nms_topk scores and rois sorted and selected
+
+    Args:
+        bounding_box_format: a case-insensitive string.
+            For detailed information on the supported format, see the
+            [KerasCV bounding box documentation](https://keras.io/api/keras_cv/bounding_box/formats/).
+        pre_nms_topk_train: int. number of top k scoring proposals to keep before applying NMS in training mode.
+            When RPN is run on multiple feature maps / levels (as in FPN) this number is per
+            feature map / level.
+        nms_score_threshold_train: float. score threshold to use for NMS in training mode.
+        nms_iou_threshold_train: float. IOU threshold to use for NMS in training mode.
+        post_nms_topk_train: int. number of top k scoring proposals to keep after applying NMS in training mode.
+            When RPN is run on multiple feature maps / levels (as in FPN) this number is per
+            feature map / level.
+        pre_nms_topk_test: int. number of top k scoring proposals to keep before applying NMS in inference mode.
+            When RPN is run on multiple feature maps / levels (as in FPN) this number is per
+            feature map / level.
+        nms_score_threshold_test: float. score threshold to use for NMS in inference mode.
+        nms_iou_threshold_test: float. IOU threshold to use for NMS in inference mode.
+        post_nms_topk_test: int. number of top k scoring proposals to keep after applying NMS in inference mode.
+            When RPN is run on multiple feature maps / levels (as in FPN) this number is per
+            feature map / level.
+
+    """
+
+    def __init__(
+        self,
+        bounding_box_format,
+        pre_nms_topk_train: int = 2000,
+        nms_score_threshold_train: float = 0.0,
+        nms_iou_threshold_train: float = 0.7,
+        post_nms_topk_train: int = 1000,
+        pre_nms_topk_test: int = 1000,
+        nms_score_threshold_test: float = 0.0,
+        nms_iou_threshold_test: float = 0.7,
+        post_nms_topk_test: int = 1000,
+        **kwargs,
+    ):
+        super().__init__(**kwargs)
+        self.bounding_box_format = bounding_box_format
+        self.pre_nms_topk_train = pre_nms_topk_train
+        self.nms_score_threshold_train = nms_score_threshold_train
+        self.nms_iou_threshold_train = nms_iou_threshold_train
+        self.post_nms_topk_train = post_nms_topk_train
+        self.pre_nms_topk_test = pre_nms_topk_test
+        self.nms_score_threshold_test = nms_score_threshold_test
+        self.nms_iou_threshold_test = nms_iou_threshold_test
+        self.post_nms_topk_test = post_nms_topk_test
+        self.built = True
+
+    def call(
+        self,
+        multi_level_boxes: Mapping[str, tf.Tensor],
```

Comment:
Should we unify the design here for the key of the mapping with the one in my PR for FPN? eg i was using the int key as the level, so that we can sort this properly (string might have the issue for '10' comes before '2'), and also using the key for loops.

Reply:
Ok, in that case, let's unify with what you have, i.e., int instead of str. Will update.

Reply:
Done.